climada/entity/exposures/base.py

```diff
@@ -424,16 +426,12 @@ def assign_centroids(self, hazard, distance='euclidean',
                     str(self.gdf.shape[0]), str(hazard.centroids.size))
         if not u_coord.equal_crs(self.crs, hazard.centroids.crs):
             raise ValueError('Set hazard and exposure to same CRS first!')
-        if hazard.centroids.meta:
-            assigned = u_coord.assign_grid_points(
-                self.gdf.longitude.values, self.gdf.latitude.values,
-                hazard.centroids.meta['width'], hazard.centroids.meta['height'],
-                hazard.centroids.meta['transform'])
-        else:
-            assigned = u_coord.assign_coordinates(
-                np.stack([self.gdf.latitude.values, self.gdf.longitude.values], axis=1),
-                hazard.centroids.coord, distance=distance, threshold=threshold)
-        self.gdf[centr_haz] = assigned
+
+        assigned_centr = u_coord.assign_gdf_centroids(self.gdf, hazard.centroids,
+                        distance=distance,
+                        threshold=threshold,test_crs=False)
```

Comment:
The problem is that climada doesn't require exp.gdf to have a valid geometry column (only a latitude and longitude column). Thus calling exp.gdf.crs throws an Error in many of the tests. (See my previous [comment](https://github.com/CLIMADA-project/climada_python/pull/602#discussion_r1093375056)).
That's why I came up with the test_crs flag. It may not be the most elegant, but do you think we should keep it? Or do you have an alternative idea?

Reply:
I see. The case of an exposures that does not have a geometry column in its gdf but who's centroids are assigned - is it a common use case? or at least valid? or just a test artefact?